Make docker push quiet in builders

soltysh opened the pull request:
@bparees @mfojtik I've completely removed the logging, adding only information about push start. Eventual push errors are logged at higher level (in `sti.go` and `docker.go`) as build failures that's why I decided not to repeat it here.

bparees commented:
hm, so now we lose the push output entirely, as opposed to being able to log it when requested?

mfojtik commented:
you should use loglevel that users might provide in Env var. also in case that push fail, we should dump the push logs in to stdout

soltysh commented:
I'm not sure if we actually loose anything other than:
```
Image 62bf5e86da53 already pushed, skipping
Image 511136ea3c5a already pushed, skipping
Image 5b12ef8fd570 already pushed, skipping
Image 684523007d12 already pushed, skipping
Image ed8e33c174f4 already pushed, skipping
...
Buffering to disk   553 kB
Buffering to disk 1.093 MB
Buffering to disk 1.625 MB
Buffering to disk 2.152 MB
Buffering to disk 2.679 MB
Buffering to disk 3.205 MB
...
Pushing [>                                                  ] 154.6 kB/13.19 MB 13s
Pushing [=>                                                 ] 315.9 kB/13.19 MB 7s
Pushing [=>                                                 ] 465.4 kB/13.19 MB 5s
Pushing [==>                                                ] 618.5 kB/13.19 MB 4s
Pushing [==>                                                ]   770 kB/13.19 MB 3s
Pushing [===>                                               ] 911.9 kB/13.19 MB 3s
Pushing [====>                                              ]  1.06 MB/13.19 MB 2s
...
```
All the other information I've added log with information about what we're pushing. In the worst case of an error builder write this to stdout as a result of a build. (see [here](https://github.com/openshift/origin/blob/master/pkg/build/builder/cmd/builder.go#L62), [here](https://github.com/openshift/origin/blob/master/pkg/build/builder/sti.go#L77) and [here](https://github.com/openshift/origin/blob/master/pkg/build/builder/docker.go#L88)).
The other option is to filter above as this does not provide any useful information, even at the highest loglevel above output is garbage IMHO.

smarterclayton commented:
We should probably have a message in the build output with "pushing" (if we don't already) because it's going to take a very long time.

```
....
Pushing image to <output>
Build complete
```

soltysh commented:
Yup, I've added that, because of that reason you've mentioned (long push time) I've added as part of this PR, see [here](https://github.com/openshift/origin/pull/1498/files#diff-1af9f534a92f3f353b73f0ed0c3a0a50R22).

smarterclayton commented on pkg/build/builder/dockerutil.go:
```diff
@@ -17,11 +19,11 @@ type DockerClient interface {
 
 // pushImage pushes a docker image to the registry specified in its tag
 func pushImage(client DockerClient, name string, authConfig docker.AuthConfiguration) error {
+	glog.Infof("Pushing %s", name)
```
`Pushing %s  ...`

soltysh replied:
Initially I wrote it with dots but since we're not using it too often I
removed them, per your request they're back.

On 10:41pm, Mon, Mar 30, 2015 Clayton Coleman <notifications@github.com>
wrote:

> In pkg/build/builder/dockerutil.go
> <https://github.com/openshift/origin/pull/1498#discussion_r27429699>:
>
> > @@ -17,11 +19,11 @@ type DockerClient interface {
> >
> >  // pushImage pushes a docker image to the registry specified in its tag
> >  func pushImage(client DockerClient, name string, authConfig docker.AuthConfiguration) error {
> > +	glog.Infof("Pushing %s", name)
>
> Pushing %s ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/1498/files#r27429699>.
>

smarterclayton commented:
We'll do a big cleanup to add dots for long running operations before we ship to be consistent with RHC (it's always `message<space>...` not dots next to text).

smarterclayton commented:
LGTM [merge]

smarterclayton commented:
[merge]
